plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcRecordCursor.java

```diff
@@ -135,8 +143,19 @@ public boolean advanceNextPosition()
 
         try {
             if (resultSet == null) {
-                log.debug("Executing: %s", statement.toString());
-                resultSet = statement.executeQuery();
+                try {
+                    resultSet = resultSetFuture.get();
```

Comment:
Asynchronous execution of statement is a non-intentional side effect.
And is also wrong, since now there is no control over how many such queries are being executed concurrently.

Move `executor.submit` here.

Also, add a comment why we're doing all this.